[Visual Refresh] Add behindText vis color tokens

mgadewoll opened the pull request:
## Summary

>[!NOTE]
This PR merges into a feature branch.

This PR adds `_behindText` vis color tokens. These have not been part of the theme, as they are not in usage in the new theme.
During upgrades to the new theme, we realized that it's rather cumbersome for teams to migrate these JSON token usages properly to the `euiTheme` tokens without a 1:1 equivalent.

These tokens are temporary for the purpose of making migration easier for Kibana teams. We will need to remove those tokens again at some point as they are not be used in Borealis.

```ts
euiColorVisBehindText0
euiColorVisBehindText1
euiColorVisBehindText2
euiColorVisBehindText3
euiColorVisBehindText4
euiColorVisBehindText5
euiColorVisBehindText6
euiColorVisBehindText7
euiColorVisBehindText8
euiColorVisBehindText9
```

## QA

These tokens are not used anywhere. Make sure the values make sense:
- Amsterdam - uses values returned from `euiPaletteColorBlindBehindText()`
- Borealis maps to default vis colors

- [x] CI passes

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8253/
- Storybook: https://eui.elastic.co/pr_8253/storybook
- New docs: https://eui.elastic.co/pr_8253/new-docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3280)
* Commit: 7528314a865be6d867a838747aa2386a9562db30

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3280","number":3280,"commit":"7528314a865be6d867a838747aa2386a9562db30"}],"number":3280}
buildkite-pr-comment-->

weronikaolejniczak commented on packages/eui-theme-common/src/global_styling/variables/colors.ts:
```diff
@@ -266,6 +266,27 @@ export type _EuiThemeVisColors = {
   euiColorVis8: string;
   euiColorVis9: string;
 
+  /* deprecated - temp token; used only during theme migration */
```
**non-blocking suggestion:** How about we use the JSDoc [@deprecated](https://jsdoc3.vercel.app/tags/deprecated) annotation for a strikethrough highlighting?

mgadewoll replied:
Yes, we should! I see how my brain was not fully there when adding this, it seems 😅 Thanks for the catch! 🙏

weronikaolejniczak commented on packages/eui/src/themes/amsterdam/global_styling/variables/_colors_vis.ts:
```diff
@@ -62,6 +62,17 @@ export const colorVis: _EuiThemeVisColors = {
   euiColorVis8: euiPaletteColorBlind.euiColorVis8.graphic,
   euiColorVis9: euiPaletteColorBlind.euiColorVis9.graphic,
 
+  euiColorVisBehindText0: '#6dccb1',
```
**non-blocking suggestion:** I understand that this is _temporary_ but for better discoverability, consistency and clarity, can we add them in `packages/eui/src/themes/amsterdam/global_styling/variables/_colors_vis.ts` and use as ☝🏻 instead of hard-coding here? We could copy-paste them from `packages/eui/src/themes/amsterdam/global_styling/variables/_colors_vis.scss`.

We could also add a comment above them, similar to `temp token; used only during theme migration`.

mgadewoll commented:
ℹ️ FYI I realized we have a breaking test due to the new year. I added the fix here on `main`: https://github.com/elastic/eui/pull/8258

I'll add the fix to this PR to be able to add the changes, but I will wait until the actual fix PR is reviewed and merged to rebase Borealis.

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8253/
- Storybook: https://eui.elastic.co/pr_8253/storybook
- New docs: https://eui.elastic.co/pr_8253/new-docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3290)
* Commit: 67971c81128a4214217c5a01051efef69aa2e497

### History
* :green_heart: [Build #3280](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3280) succeeded 7528314a865be6d867a838747aa2386a9562db30

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3290","number":3290,"commit":"67971c81128a4214217c5a01051efef69aa2e497"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3280","number":3280,"commit":"7528314a865be6d867a838747aa2386a9562db30"}],"number":3290}
buildkite-pr-comment-->

mgadewoll commented:
> ℹ️ FYI I realized we have a breaking test due to the new year. I added the fix here on `main`: #8258
> 
> I'll add the fix to this PR to be able to add the changes, but I will wait until the actual fix PR is reviewed and merged to rebase Borealis.

ℹ️ I rebased this branch with the updated Borealis feature branch that includes the test pipeline fix.

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8253/
- Storybook: https://eui.elastic.co/pr_8253/storybook
- New docs: https://eui.elastic.co/pr_8253/new-docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3298)
* Commit: e1109c1552ccb47648159e97e8037687b1e5ea83

### History
* :green_heart: [Build #3290](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3290) succeeded 67971c81128a4214217c5a01051efef69aa2e497
* :green_heart: [Build #3280](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3280) succeeded 7528314a865be6d867a838747aa2386a9562db30

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3298","number":3298,"commit":"e1109c1552ccb47648159e97e8037687b1e5ea83"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3290","number":3290,"commit":"67971c81128a4214217c5a01051efef69aa2e497"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3280","number":3280,"commit":"7528314a865be6d867a838747aa2386a9562db30"}],"number":3298}
buildkite-pr-comment-->
